backend_jobs/verify_labels/pipeline_lib/firestore_database.py

```diff
@@ -0,0 +1,169 @@
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+"""
+
+import apache_beam as beam
+
+from backend_jobs.pipeline_utils.firestore_database import initialize_db, RANGE_OF_BATCH
+from backend_jobs.pipeline_utils import database_schema, data_types
+from backend_jobs.recognition_pipeline.pipeline_lib.firestore_database import add_id_to_dict
+
+# pylint: disable=abstract-method
+class GetBatchedLabelsDataset(beam.DoFn):
+    """Queries the project's database to get the labels needed to be verified.
+
+    Input:
+       integer index between 0 and 9.
+
+    Output:
+        generator of label's documents in a Python dictionary form.
+        Each label is represented by a dict containing all the fields
+        of the document in the database and their values as stored in
+        the database_schema.COLLECTION_IMAGES_SUBCOLLECTION_LABELS.
+
+    """
+    def setup(self):
+        # pylint: disable=attribute-defined-outside-init
+        self.db = initialize_db()
+
+    # pylint: disable=arguments-differ
+    def process(self, element, recognition_run):
+        """Queries firestore database for labels recognizied by
+        the recognition_run within a random range (by batch).
+
+        Args:
+            element: the lower limit for querying the database by the random field.
+            recognition_run: the input of the pipeline, determines the labels dataset.
+
+        Returns:
+            A generator of dictionaries with all the information (fields and id)
+            of each one of the Firestore data set's label documents as stored in
+            the database_schema.COLLECTION_IMAGES_SUBCOLLECTION_LABELS.
+
+        """
+        # the lower limit for querying the database by the random field.
+        random_min = element * RANGE_OF_BATCH
+        # the higher limit for querying the database by the random field.
+        random_max = random_min + RANGE_OF_BATCH
+        query = self.db.collection_group(database_schema.COLLECTION_IMAGES_SUBCOLLECTION_LABELS)\
+            .where(database_schema.COLLECTION_IMAGES_SUBCOLLECTION_LABELS_FIELD_PIPELINE_RUN_ID,\
+                u'==', recognition_run).where(database_schema.\
+                    COLLECTION_IMAGES_SUBCOLLECTION_LABELS_FIELD_RANDOM, u'>=', random_min)\
+                        .where(database_schema.COLLECTION_IMAGES_SUBCOLLECTION_LABELS_FIELD_RANDOM,\
+                            u'<', random_max).stream()
+        return (add_id_to_dict(doc) for doc in query)
+
+# pylint: disable=abstract-method
+class UpdateDatabase(beam.DoFn):
```

Comment:
Change to a more descriptive name

Reply:
changed to UpdateDatabaseWithVisibleLabels